reseq_ros2/reseq_ros2/communication.py

```diff
@@ -41,11 +45,30 @@ def __init__(self):
                 bustype="socketcan",
             )
             self.notifier = can.Notifier(self.canbus, [self.can_callback])
-        except OSError:
+        except OSError as e:
+            self.get_logger().fatal(f'Error while connecting to CAN bus: {str(e)}\n{traceback.format_exc()}')
             raise
 
         self.get_logger().info("Communication node started")
 
+        # Start the CAN bus monitoring thread
+        self.active = True
+        self.monitoring_thread = threading.Thread(target=self.monitor_canbus)
+        self.monitoring_thread.start()
+
+    def monitor_canbus(self):
+        process = subprocess.Popen(["candump", self.can_channel], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        while self.active:
+            retcode = process.poll()  # Check if the candump process has terminated
+            if retcode is not None:  # If the process has terminated
+                if retcode != 0:
+                    self.get_logger().fatal(f'candump process terminated with return code {retcode}')
+                self.active = False
+                self.exit_code = retcode # Save the retcode for later use
+                process.terminate()
+                rclpy.shutdown()
+                return
+
```

Comment:
I didn't look, as I mentioned in my previous comments, to the complete PR and actually I do not understand what this function is trying to obtain.

From what I understand both the `candump` and `SocketCan` are only looking at the presence of the `can` socket interface. While sometimes (if we forget or if there are some errors) it is possible that the socket is not present at startup (which cause the `OSError` we wanted to deal with, I cannot see why it would suddenly disappear. 

If we are interested in performing some checks on the CAN I would instead suggest the script that I created (`test_can.py`) which actually verifies that we are receiving the messages we need, however I think that this should be discussed further

Unless there is some reasons for this that I am missing we can remove this monitor and all the problems you listed with the Exception handling should be solved

Reply:
Yes, this is the reason why I created the CAN monitor function, as one of my tasks was to crash the communication if CAN crashed. But it is now unnecessary as you have created a script that does a similar job.